src/DataStructures/MakeWithValue.hpp

```diff
@@ -0,0 +1,42 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+///\file
+/// Defines make_with_value
+
+#pragma once
+
+#include "Utilities/ForceInline.hpp"
+
+namespace MakeWithValueImpl_detail {
+template <typename R, typename T, typename V>
+struct MakeWithValueImpl {
+  static SPECTRE_ALWAYS_INLINE R apply(const T& input, const V& value);
+};
+}  // namespace MakeWithValueImpl_detail
+
+/// \ingroup DataStructures
+/// \brief Given an object of type `T`, create an object of type `R` whose
+/// elements are initialized to `value`.
+///
+/// \details This function is useful in templates in order to initialize the
+/// return type of a function template with `value` for functions that can be
+/// called either at a single grid-point or to fill a data structure at the same
+/// set of grid-points as the `input`
+///
+/// \see MakeWithValueImpl
```

Comment:
since MakeWithValueImpl is in a `_detail` namespace this won't link to anything. I'm not really sure what the right change is, but I wanted to mention it. Of course, if I'm wrong please correct me :)